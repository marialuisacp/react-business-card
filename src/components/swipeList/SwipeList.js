import React, { Component } from 'react';
import ReactSwipe from 'react-swipe';
import PropTypes from 'prop-types';

import ListPeople from '../listPeople/ListPeople';

import './SwipeList.scss';
import '../../styles/styles.scss';

class SwipeList extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    let reactSwipeEl;
    const { data } = this.props;
    return (
      <div className='component' id='swipe-list-component'>
        <div>
          <ReactSwipe
            className="carousel"
            swipeOptions={{ continuous: true }}
            ref={el => (reactSwipeEl = el)}
          >
            <div>
              <ListPeople data={data.prev.people} />
            </div>
            <div>
              <ListPeople data={data.current.people} />
            </div>
            <div>
              <ListPeople data={data.next.people} />
            </div>
          </ReactSwipe>
          <button onClick={() => reactSwipeEl.next()}>Next</button>
          <button onClick={() => reactSwipeEl.prev()}>Previous</button>
        </div>
      </div >
    );
  }
}

SwipeList.propTypes = {
  data: PropTypes.object
};

export default SwipeList;