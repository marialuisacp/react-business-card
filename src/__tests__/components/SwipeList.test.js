import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import SwipeList from '../../components/SwipeList/SwipeList';
import { Provider } from 'react-redux'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('SwipeList', () => {
  test('Component SwipeList should render without crashing given the required props', () => {
    const props = {
      letterCenter: 'D',
      dispatch: jest.fn(),
      direction: 1
    }

    const store = mockStore({
      currentLetter: 'D',
      direction: 0
    })

    const wrapper = shallow(<Provider store={store}>
      <SwipeList {...props} />
    </Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})