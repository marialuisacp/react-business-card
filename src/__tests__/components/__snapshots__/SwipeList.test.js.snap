// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeList Component SwipeList should render without crashing given the required props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(SwipeList)
    direction={1}
    dispatch={[MockFunction]}
    letterCenter="D"
  />
</ContextProvider>
`;
